operators/src/processing/interpolation/mod.rs

```diff
@@ -0,0 +1,824 @@
+            params: self.params,
+            tiling_specification: context.tiling_specification(),
+        };
+
+        Ok(initialized_operator.boxed())
+    }
+}
+
+pub struct InitializedInterpolation {
+    result_descriptor: RasterResultDescriptor,
+    raster_source: Box<dyn InitializedRasterOperator>,
+    params: InterpolationParams,
+    tiling_specification: TilingSpecification,
+}
+
+impl InitializedRasterOperator for InitializedInterpolation {
+    fn query_processor(&self) -> Result<TypedRasterQueryProcessor> {
+        let source_processor = self.raster_source.query_processor()?;
+
+        let res = call_on_generic_raster_processor!(
+            source_processor, p => match self.params.interpolation  {
+                InterpolationMethod::NearestNeighbor => InterploationProcessor::<_,_, NearestNeighbor>::new(
+                        p,
+                        self.params.clone(),
+                        self.tiling_specification,
+                        self.result_descriptor.no_data_value.unwrap().as_(),
+                    ).boxed()
+                    .into(),
+                InterpolationMethod::BiLinear =>InterploationProcessor::<_,_, Bilinear>::new(
+                        p,
+                        self.params.clone(),
+                        self.tiling_specification,
+                        self.result_descriptor.no_data_value.unwrap().as_(),
+                    ).boxed()
+                    .into(),
+            }
+        );
+
+        Ok(res)
+    }
+
+    fn result_descriptor(&self) -> &RasterResultDescriptor {
+        &self.result_descriptor
+    }
+}
+
+pub struct InterploationProcessor<Q, P, I>
+where
+    Q: RasterQueryProcessor<RasterType = P>,
+    P: Pixel,
+    I: InterpolationAlgorithm<P>,
+{
+    source: Q,
+    params: InterpolationParams,
+    tiling_specification: TilingSpecification,
+    no_data_value: P,
+    interpolation: PhantomData<I>,
+}
+
+impl<Q, P, I> InterploationProcessor<Q, P, I>
+where
+    Q: RasterQueryProcessor<RasterType = P>,
+    P: Pixel,
+    I: InterpolationAlgorithm<P>,
+{
+    pub fn new(
+        source: Q,
+        params: InterpolationParams,
+        tiling_specification: TilingSpecification,
+        no_data_value: P,
+    ) -> Self {
+        Self {
+            source,
+            params,
+            tiling_specification,
+            no_data_value,
+            interpolation: PhantomData,
+        }
+    }
+}
+
+#[async_trait]
+impl<Q, P, I> QueryProcessor for InterploationProcessor<Q, P, I>
+where
+    Q: QueryProcessor<Output = RasterTile2D<P>, SpatialBounds = SpatialPartition2D>,
+    P: Pixel,
+    I: InterpolationAlgorithm<P>,
+{
+    type Output = RasterTile2D<P>;
+    type SpatialBounds = SpatialPartition2D;
+
+    async fn query<'a>(
+        &'a self,
+        query: RasterQueryRectangle,
+        ctx: &'a dyn QueryContext,
+    ) -> Result<BoxStream<'a, Result<Self::Output>>> {
+        // TODO: check input resolution < output resolution
```

Comment:
please add that.

Reply:
9fba3541e7ffdc6b094d5a270c28ddcc36909909